test/augmentation/test_augmentation.py

```diff
@@ -25,6 +25,334 @@
+
+        # Output should match
+        assert output.shape == expected_output.shape
+        assert_allclose(output, expected_output.to(device).to(dtype), atol=1e-4, rtol=1e-4), output
+
+        # Transformation should match
+        assert transformation.shape == expected_transformation.shape
+        assert_allclose(transformation, expected_transformation.to(
+            device).to(dtype), atol=1e-4, rtol=1e-4), transformation
+
+    # TODO finish it
+
+    def _test_random_batch_transform_implementation(
+            self, device, dtype, input_tensor, expected_output, expected_transformation, batch_prob, **params):
+        augmentation = self._create_augmentation_from_params(**params, p=1.0, return_transform=True)
+        forward_params = {**augmentation.generate_parameters(input_tensor.shape), "batch_prob": batch_prob}
+        output, transformation = augmentation(input_tensor.to(device).to(dtype), params=forward_params)
+
+        # Output should match
+        assert output.shape == expected_output.shape
+        assert_allclose(output, expected_output.to(device).to(dtype), atol=1e-4, rtol=1e-4), output
+
+        # Transformation should match
+        assert transformation.shape == expected_transformation.shape
+        assert_allclose(transformation, expected_transformation.to(
+            device).to(dtype), atol=1e-4, rtol=1e-4), transformation
+
+    # TODO finish it
+
+    def _test_sequential(self, device, dtype, **params):
+
+        f = nn.Sequential(
+            self._create_augmentation_from_params(**params, return_transform=True),
+            self._create_augmentation_from_params(**params, return_transform=True)
+        )
+
+        input = torch.rand(3, 5, 5).to(device)  # 3 x 5 x 5
+
+        expected = input
+
+        expected_transform = torch.eye(3).unsqueeze(0)  # 3 x 3
+        expected_transform = expected_transform.to(device)
+
+        assert_allclose(f(input)[0], expected, atol=1e-4, rtol=1e-5)
+        assert_allclose(f(input)[1], expected_transform)
+
+
+class TestColorJitterAlternative(CommonTests):
+
+    possible_params = {
+        "brightness": (1., (0., 2.), [0., 2.], torch.tensor(0.), torch.tensor((0., 2.))),
+        "contrast": (1., (0., 2.), [0., 2.], torch.tensor(0.), torch.tensor((0., 2.))),
+        "saturation": (1., (0., 2.), [0., 2.], torch.tensor(0.), torch.tensor((0., 2.))),
+        "hue": (0., (-0.5, 0.5), [-0.5, 0.5], torch.tensor(0.), torch.tensor((-0.5, 0.5))),
+    }
+    params = [{param_name: v} for param_name, possible_values in possible_params.items() for v in possible_values]
+
+    _augmentation_cls = ColorJitter
+    _default_param_set = {"brightness": 0.3,
+                          "contrast": 0.3,
+                          "saturation": 0.3,
+                          "hue": 0.3, }
+
+    @pytest.fixture(params=params, scope="class")
+    def param_set(self, request):
+        return request.param
```

Comment:
what's the use for this `param_set` fixture ?

Reply:
This fixture is used in the smoke tests in the CommonTests, one has to override it to be able to reparametrise it I think. This way one doesn't have to override and call here the smoke tests or any other test already implemented in the CommonTests, just if it doesn't fully fit in, f.e. ColorJitter requires 3 channels, so doesn't work for  test_consistent_output_shape with input shape (4,5)